internal/checker/checker.go

```diff
@@ -522,3 +536,48 @@ func (c *Checker) ReadPacks(ctx context.Context, packs map[restic.ID]int64, p *p
 		}
 	}
 }
+
+// Find data packfiles for repository checking  based on snapshots.
+// Use restic.StreamTrees to gather all data blobs and convert them to their
+// containing packfile
+func (c *Checker) FindDataPackfiles(ctx context.Context, repo *repository.Repository, sn *restic.Snapshot,
```

Comment:
Change this to receive _all_ relevant snapshots at once. Then there's no need for `c.packSet`.